plugin/core/rpc.py

```diff
@@ -296,7 +296,8 @@ def deduce_payload(
             response_id = int(payload["id"])
             handler, result = self.response_handler(response_id, payload)
             response_tuple = (handler, result, None, None, None)
-            self.logger.incoming_response(response_id, result)
+            blocking = self._sync_request_result.is_ready()
```

Comment:
This condition might be unclear on first sight. I could also compare `response_id` with `request_id` if that would make it clearer.

Reply:
You are returning `result` when `is_requesting()` in `response_handler`, but here we check for `is_ready()`. I don't understand this.

Reply:
`set()` or `set_error()` is called after `is_requesting()`, before returning. So that makes it `is_ready()`.

Reply:
I see, good point!